streams/src/test/java/org/apache/kafka/streams/KafkaStreamsTest.java

```diff
@@ -661,6 +673,131 @@ public void statefulTopologyShouldCreateStateDirectory() throws Exception {
+
+
+
+        final Serializer mockOutputKeySerializer = mock(Serializer.class);
+        mockOutputKeySerializer.configure(anyObject(), eq(true));
+        expectLastCall();
+        final Serializer mockOutputValueSerializer = mock(Serializer.class);
+        mockOutputValueSerializer.configure(anyObject(), eq(false));
+        expectLastCall();
+
+        final Serde mockOutputKeySerde = mock(Serde.class);
+        final Serde mockOutputValueSerde = mock(Serde.class);
+        expect(mockOutputKeySerde.serializer()).andReturn(mockOutputKeySerializer).anyTimes();
+        expect(mockOutputValueSerde.serializer()).andReturn(mockOutputValueSerializer).anyTimes();
+
+
+
+        final Deserializer mockGlobalKeyDeserializer = mock(Deserializer.class);
+        mockGlobalKeyDeserializer.configure(anyObject(), eq(true));
+        expectLastCall();
+        final Deserializer mockGlobalValueDeserializer = mock(Deserializer.class);
+        mockGlobalValueDeserializer.configure(anyObject(), eq(false));
+        expectLastCall();
+
+        final Serde mockGlobalKeySerde = mock(Serde.class);
+        final Serde mockGlobalValueSerde = mock(Serde.class);
+        expect(mockGlobalKeySerde.deserializer()).andReturn(mockGlobalKeyDeserializer).anyTimes();
+        expect(mockGlobalValueSerde.deserializer()).andReturn(mockGlobalValueDeserializer).anyTimes();
+
+
+
+        builder
+            .stream("anyTopic", Consumed.with(mockSourceKeySerde, mockSourceValueSerde))
+            .through("anyOtherTopic", Produced.with(mockThroughKeySerde, mockThroughValueSerde))
+            .selectKey(KeyValue::pair)
+            .groupByKey(Grouped.with(mockGroupedKeySerde, mockGroupedValueSerde))
+            .count()
+            .toStream()
+            .to("anyOutput", Produced.with(mockOutputKeySerde, mockOutputValueSerde));
+        builder.globalTable("anyGlobal", Consumed.with(mockGlobalKeySerde, mockGlobalValueSerde));
+
+        replay(
+            mockSourceKeyDeserialzer, mockSourceValueDeserialzer, mockSourceKeySerde, mockSourceValueSerde,
+            mockThroughKeySerializer, mockThroughKeyDeserializer, mockThroughKeySerde,
+            mockThroughValueSerializer, mockThroughValueDeserializer, mockThroughValueSerde,
+            mockGroupedKeySerializer, mockGroupedKeyDeserializer, mockGroupedKeySerde,
+            mockGroupedValueSerializer, mockGroupedValueDeserializer, mockGroupedValueSerde,
+            mockOutputKeySerializer, mockOutputValueSerializer, mockOutputKeySerde, mockOutputValueSerde,
+            mockGlobalKeyDeserializer, mockGlobalValueDeserializer, mockGlobalKeySerde, mockGlobalValueSerde);
+
+        KafkaStreams kafkaStreams = null;
+        try {
+            kafkaStreams = new KafkaStreams(builder.build(), props);
+        } finally {
+            if (kafkaStreams != null) {
+                kafkaStreams.close();
+            }
+        }
+
+        verify(
```

Comment:
Why we do not want to verify serde classes here? Although it's trivial seems no harm either.

Reply:
Well. What we want to test is, that `Deserializer#configure()` and `Serializer#configure()` are called. Note, that `Serde#configure()` is not called because we extract the `Deserializer` and `Serializer` from the `Serdes`. Thus, it seems to be a distraction of the test purpose.

If we would verify the serdes, we would only verify that `Serde#serializer()` or `Serde#deserializer()` are called. But we know that anyway, because otherwise it's impossible to call `configure()` on the wrapped (de)serializers. Also, technical this test does not care if those method are called.

Reply:
Ack, that makes sense.